plugins/modules/event_info.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Paul Arthur <paul.arthur@flowerysong.com>
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'XLAB Steampunk'}
+
+DOCUMENTATION = '''
+module: event_info
+author:
+  - Paul Arthur (@flowerysong)
+  - Aljaz Kosir (@aljazkosir)
+  - Manca Bizjak (@mancabizjak)
+  - Tadej Borovsak (@tadeboro)
+short_description: Lists Sensu events
+description:
+  - For more information, refer to the Sensu documentation at
+    U(https://docs.sensu.io/sensu-go/latest/reference/events/)
+extends_documentation_fragment:
+  - sensu.sensu_go.base
+options:
+  check_name:
+    description:
+      - Limit results to a specific check.
+      - C(entity_name) must also be specified.
+    type: str
+  entity_name:
+    description:
+      - Limit results to a specific entity.
+    type: str
+'''
+
+EXAMPLES = '''
+- name: List Sensu events
+  event_info:
+  register: result
+
+- name: List Sensu events for api.example.com
+  event_info:
+    entity: api.example.com
+  register: result
+'''
+
+RETURN = '''
+objects:
+  description: list of Sensu events
+  returned: always
+  type: list
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils,
+)
+
+
+def main():
+    required_by = {'check_name': ['entity_name']}
+    module = AnsibleModule(
+        supports_check_mode=True,
+        required_by=required_by,
+        argument_spec=dict(
+            arguments.get_spec("auth"),
+            check_name=dict(),
+            entity_name=dict(),
+        ),
+    )
+
+    client = arguments.get_sensu_client(module.params['auth'])
+    if module.params['check_name']:
+        path = '/events/{0}/{1}'.format(module.params['entity_name'], module.params['check_name'])
+    elif module.params['entity_name']:
+        path = '/events/{0}'.format(module.params['entity_name'])
+    else:
+        path = '/events'
```

Comment:
Considering that the `/events` part of the path will always be present, I suggest removing `else` and move this line before L77. Then, if necessarry (i.e. depending on the provided params), you can append the appropriate trailing part of the path.

This way we can reduce `if-elif-else` to `if-elif` :slightly_smiling_face:

Reply:
Or we can go almost completely `if`-less with

    path_components = '/events', module.params['entity'], module.params['check']
    path = '/'.join(c for c in path_components if c)

But this is just me golfing, so feel free to ignore my code ;)

Reply:
That's way cooler! :+1: